doc/dev/design/NAT-address-discovery.rst

```diff
@@ -0,0 +1,143 @@
+    NATed IP/port).
+
+    -  Advantages: One less dependency on an external library and protocol
+    -  Disadvantages: Conceptually a bit of a hack. More standardization effort? How do we solve authentication?
+
+
+Rationale
+=========
+[A discussion of alternate approaches and the trade-offs, advantages, and disadvantages of the specified approach.]
+
+The main reasons for integrating the functionality with the BR are:
+
+-  Reliability: The border router is almost guaranteed to see the correct IP/port on the NAT, especially when using the
+   same port for STUN traffic and routing traffic. All other approaches rely on the leniency of the NAT to use the same
+   port even if the STUN server and border router have different ports or even IPs.
+-  Time to rollout: changing the border routers should be much easier and faster than getting router vendors or NAT
+   vendors to implement SCION compatibility or to get rid of NATs completely in home networks.
+   A short time until rollout seems important because people are already running into this problem.
+
+Alternatives:
+
+-  The SRC address/port is updated by the border router to reflect what the border router sees as source address.
+   Problem:
+   -  Complicates cryptographically protecting the header if the header must be modifiable by border routers
+   -  A spoofed IP causes the border router to route traffic to an unsuspecting target
+-  The SRC address/port is updated by the NAT. This is similar to having the border router update tSRC IP/port.
+   - Complicates cryptographically protecting the header if the header must be modifiable by border routers
+-  Use STUN servers. This is a possibility, but adds setup complexity and may not work in all cases. Every subnet
+   of an AS that has a border router must also have a STUN server. Moreover, if the STUN server uses a different IP
+   (or port) than the border router, then the NAT may decide to use a different port when connecting to it, i.e. the
+   STUN server may not see the same IP/port tuple on the NAT that the border router sees. Disadvantages:
+
+   -  This approach may be be problematic with sensitive NATs.
+   -  We need to somehow standardize the STUN IP/port and/or communicate it to endhosts, e.g. via the topo file or
+      its successor.
+-  Remove all NATs and use IPv6 instead. This is technically possible but unlikely to happen anytime soon.
+
+Compatibility
+=============
+[A discussion of breaking changes and how this change can be deployed.]
+
+Breaking changes
+----------------
+
+This change should not break anything.
+
+Transition
+----------
+
+- An "old" client without expectation on STUN support on the router would simply not use this feature. No problem here.
+- A "new" client sending a STUN request to an "old" border router should simply fail because the router should simply
+  drop a packet that it cannot process.
+  The client should then time out and report that the external NAT address cloud not be established.
```

Comment:
Wouldn't it be better in this case to maybe log the above fact but instead of failing to optimistically assume that no NAT is deployed?

Reply:
I added that as an alternative with `TBD`. You are probably right, but I think some people may prefer erroring out because it clearly indicates an outdated border router installation.